pkg/metrics/metrics_recorder_test.go

```diff
@@ -0,0 +1,205 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	risingwavev1alpha1 "github.com/risingwavelabs/risingwave-operator/apis/risingwave/v1alpha1"
+	"github.com/risingwavelabs/risingwave-operator/pkg/utils"
+)
+
+type panicValWebhook struct{}
+
+func (p *panicValWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (err error) {
+	panic("validateCreate panic")
+}
+
+func (p *panicValWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) error {
+	panic("validateDelete update")
+}
+
+func (p *panicValWebhook) ValidateUpdate(ctx context.Context, oldObj runtime.Object, newObj runtime.Object) (err error) {
+	panic("validateUpdate panic")
+}
+
+func Test_MetricsValidatingWebhookPanic(t *testing.T) {
+	ResetMetrics()
+	risingwave := &risingwavev1alpha1.RisingWave{}
+	panicWebhook := NewValidatingWebhookMetricsRecorder(&panicValWebhook{})
+
+	_ = panicWebhook.ValidateCreate(context.Background(), risingwave)
+	assert.Equal(t, 1, GetWebhookRequestPanicCountWith(utils.ValidatingWebhookType, risingwave), "Panic metric")
+	assert.Equal(t, 1, GetWebhookRequestRejectCount(utils.ValidatingWebhookType, risingwave), "Reject metric")
+	assert.Equal(t, 1, GetWebhookRequestCount(utils.ValidatingWebhookType, risingwave), "Count metric")
+	assert.Equal(t, 0, GetWebhookRequestPassCount(utils.ValidatingWebhookType, risingwave), "Pass metric")
+	ResetMetrics()
+
+	_ = panicWebhook.ValidateDelete(context.Background(), risingwave)
+	assert.Equal(t, 1, GetWebhookRequestPanicCountWith(utils.ValidatingWebhookType, risingwave), "Panic metric")
+	assert.Equal(t, 1, GetWebhookRequestRejectCount(utils.ValidatingWebhookType, risingwave), "Reject metric")
+	assert.Equal(t, 1, GetWebhookRequestCount(utils.ValidatingWebhookType, risingwave), "Count metric")
+	assert.Equal(t, 0, GetWebhookRequestPassCount(utils.ValidatingWebhookType, risingwave), "Pass metric")
+	ResetMetrics()
+
+	_ = panicWebhook.ValidateUpdate(context.Background(), risingwave, risingwave)
+	assert.Equal(t, 1, GetWebhookRequestPanicCountWith(utils.ValidatingWebhookType, risingwave), "Panic metric")
+	assert.Equal(t, 1, GetWebhookRequestRejectCount(utils.ValidatingWebhookType, risingwave), "Reject metric")
+	assert.Equal(t, 1, GetWebhookRequestCount(utils.ValidatingWebhookType, risingwave), "Count metric")
+	assert.Equal(t, 0, GetWebhookRequestPassCount(utils.ValidatingWebhookType, risingwave), "Pass metric")
+	ResetMetrics()
+}
+
+type successfulValWebhook struct{}
```

Comment:
I did not have a look at the tests in detail. You just moved them, correct?

Reply:
yup, just a moving.